rasa_core/processor.py

```diff
@@ -121,14 +121,13 @@ def continue_message_handling(self, sender_id, executed_action, events):
         self._log_action_on_tracker(tracker, executed_action, events)
         if self._should_predict_another_action(executed_action, events):
             action = self._get_next_action(tracker)
-            # save tracker state to continue conversation from this state
-            self._save_tracker(tracker)
             return {"next_action": action.name(),
                     "tracker": tracker.current_state()}
         else:
             return {"next_action": None,
                     "tracker": tracker.current_state()}
-
+        # save tracker state to continue conversation from this state    
```

Comment:
Since this after the `return`s of all the branches, this will never be executed. I don't see why you moved it there.

Reply:
Oh, It's my miss, I would like to add _save_tracker in 'else' block for the event_restart and event_action_listen, I have a new commit, Sorry.